Cython/Compiler/ExprNodes.py

```diff
@@ -189,20 +189,31 @@ def infer_sequence_item_type(env, seq_node, index_node=None, seq_type=None):
 
 def get_exception_handler(exception_value):
     if exception_value is None:
-        return "__Pyx_CppExn2PyErr();"
+        return "__Pyx_CppExn2PyErr();", False
+    elif (exception_value.type == PyrexTypes.c_char_type
```

Comment:
Can this collide with an exception handler function that returns a `char`?

Reply:
The declarator is completely parsed by this point, and the type is a function type.